lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1233,6 +1238,9 @@ def _on_logging_relation_changed(self, event: HookEvent):
             event: a `CharmEvent` in response to which the consumer
                 charm must update its relation data.
         """
+        if not self.container.can_connect():
+            return
```

Comment:
I think it is worth leaving here a comment as to why this event is not deferred.

Reply:
```suggestion
            # We do not defer the event because, if it is a relation created/changed, we will
            # process the relation again on pebble_ready, and the nature of the event is not
            # taken into account in the following logic.
            return
```